### PR TITLE
Fix publish target

### PR DESCRIPTION
**Pull request by carolynvs**
I have converted the repository from make to mage, and from azure devops pipelines to github actions. I figured as long as I was here fixing our broken build, lets just finish moving over to where we want to be.

The problem with the publish target was a few things. One there was a bug in the magefiles generatePackageFeed function, where it wasn't checking the error when calling porter to generate the atom.xml file for the feed. Porter wasn't installed, and it was quietly failing.

We are now using a [fixed version of the magefiles](https://github.com/getporter/magefiles/pull/13) repo, and installing porter into bin before attempting to publish.

🚨 I will update this PR once https://github.com/getporter/magefiles/pull/13 is merged. Do not merge until its in.

### CODE REVIEW

**Comment by carolynvs** [comment]
Oops, one more change is needed so hold off on reviewing. I need to make a personal access token and use it in the workflow since the default token doesn't have access to the packages repository where we upload the updated atom feed.

**Comment by carolynvs** [comment]
Here is a successful publish that I ran on my fork

**Comment by carolynvs** [comment]
Here is a successful run based on a tag:

https://github.com/carolynvs/azure-plugins/runs/6575444361?check_suite_focus=true

https://github.com/carolynvs/azure-plugins/releases/tag/v1.0.0-beta.2

When branches are pushed to that repo, publish is skipped (we only publish canary and tagged releases using logic built into porter).

**Comment by carolynvs** [comment]
Okay! This is now ready to review. I've defined a new organization secret PUBLISH_TOKEN that all the repos can use to publish binaries and the atom feed.

File: magefile.go [hunk 1]
```diff
@@ -35,3 +58,83 @@ func TestPublish(username string) {
...
+	if rebuild {
+		mgx.Must(releases.BuildClient(pluginPkg, pluginName, binDir))
+	} else {
+		fmt.Println("target is up-to-date")
+	}
+}
+
+// XBuildAll cross-compiles the plugin
+func XBuildAll() {
+	rebuild, err := target.Path(filepath.Join(binDir, "dev/azure-linux-amd64"), srcDirs...)
```
Comment (VinozzZ) [inline magefile.go:125]: Should we use target.Dir here?

**Review by VinozzZ — APPROVED** [review]

**Comment by carolynvs** [comment]
*squeezes her eyes shut and presses the green button* 😆

**Pull request merged**